qiskit/qasm3/grammar.py

```diff
@@ -0,0 +1,644 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name, abstract-method, super-init-not-called
+
+"""QASM3 AST Nodes"""
+
+
+class Class:
```

Comment:
Is there a more detailed name we can use? `ASTNode`?

Reply:
Good one. I thought `Class` as a grammar class. But `ASTNode` is much better. Done in b5a1982469eda9e9f67dc3a8a738ff33096cf0f3